src/transformers/integrations/peft.py

```diff
@@ -312,14 +327,22 @@ def enable_adapters(self) -> None:
 
         for _, module in self.named_modules():
             if isinstance(module, BaseTunerLayer):
-                module.disable_adapters = False
+                # The recent version of PEFT need to call `enable_adapters` instead
+                if hasattr(module, "enable_adapters"):
+                    module.enable_adapters(enabled=True)
+                else:
+                    module.disable_adapters = False
 
-    def active_adapter(self) -> str:
+    def active_adapters(self) -> List[str]:
```

Comment:
Instead of doing two functions:
- `active_adapters` and `active_adapter` I would just do one function `active_adapter`

```suggestion
    def active_adapter(self) -> Union[str, List[str]]:
```

Reply:
I think that way it won't be BC --> for users that have previous PEFT version they will get an str whereas for newest PEFT versions they will get a list of str